### DISCUSSION TITLE
How to access $t() inside script?

### DISCUSSION THREAD

**Discussion by LifeIsStrange** [intlify/vue-i18n#825] [Q&A] [answered]
I can access $t() in the template of a SFC component but for some reason, it is undefined in the component script...
(I am using Vue 3 script setup and typescript and vue-i18n global scope)
t() is undefined too and this for globals no longer exist in Vue 3.

**Comment by kazupon** [comment, marked as answer]
you need to use the `useI18n` in script setup.
You can check the vue-i18n-next docs
https://vue-i18n.intlify.dev/guide/advanced/composition.html

**Reply by LifeIsStrange** [reply]
> you need to use the `useI18n` in script setup. You can check the vue-i18n-next docs https://vue-i18n.intlify.dev/guide/advanced/composition.html

@kazupon Thanks, you're amazing :D
